src/core_init_atmosphere/mpas_geotile_manager.F

```diff
@@ -0,0 +1,1025 @@
+
+        ! Stack Procedures
+        procedure, public :: push_tile => mpas_geotile_mgr_push_tile
+        procedure, public :: pop_tile => mpas_geotile_mgr_pop_tile
+        procedure, public :: is_stack_empty => mpas_geotile_mgr_stack_is_empty
+
+        ! Private Procedures
+        procedure, private :: search_tile => mpas_geotile_mgr_search_tile
+        procedure, private :: add_tile => mpas_geotile_mgr_add_tile
+        procedure, private :: gen_filename => mpas_geotile_mgr_gen_tile_name
+        procedure, private :: hash_to_ll => mpas_geotile_mgr_hash_to_latlon
+    end type mpas_geotile_mgr_type
+
+
+    type, extends(mpas_stack_payload_type) :: mpas_geotile_type
+        real (c_float), dimension(:,:,:), pointer :: tile
+
+        character (len=StrKIND) :: fname ! Path to the file that contains the data for this tile
+        integer :: hash_x ! The x offset of this tile in the hash table
+        integer :: hash_y ! The y offset of this tile in the hash table
+
+        integer :: x, y ! The tiles range, in pixels
+        logical :: is_processed = .false.
+    end type mpas_geotile_type
+
+
+    type tile_hash
+        type(mpas_geotile_type), pointer :: ptr => null()
+    end type tile_hash
+
+
+    contains
+
+
+    !***********************************************************************
+    !
+    !  public function mpas_geotile_mgr_init => init
+    !
+    !> \brief   Initialize a mpas_geotile_mgr class
+    !> \author  Miles A. Curry
+    !> \date    02/20/2020
+    !> \details
+    !> Initialize a geotile manager class by parsing the index file located
+    !> within path and allocated needed data structures for static interpolation.
+    !> Init should be called before calling any other mpas_geotile_mgr_type
+    !> procedures. If path is not a directory or no index file is found in path,
+    !> 1 will be returned. Upon success 0 will be returned.
+    !
+    !-----------------------------------------------------------------------
+    function mpas_geotile_mgr_init(mgr, path) result(ierr)
+
+        use mpas_parse_geoindex, only : mpas_parse_index
+
+        implicit none
+
+        ! Input variables
+        class (mpas_geotile_mgr_type) :: mgr
+        character (len=*), intent(in) :: path
+
+        ! Local variables
+        character (len=StrKIND), pointer :: endian
+        integer, pointer :: tile_nx ! Number of pixels in the x-direction for a single tile
+        integer, pointer :: tile_ny ! Number of pixels in the y-direction for a single tile
+        integer, pointer :: signed
+        integer :: err_level
+        real (kind=RKIND), pointer :: dx ! Grid spacing in the x-direction
+        real (kind=RKIND), pointer :: dy ! Grid spacing in the y-direction
+        real (kind=RKIND), pointer :: scalefactor
+        logical :: res
+
+        ! Return variable
+        integer :: ierr
+
+        ierr = 0
+
+        ! Check to see if the directory exists
+        inquire(file=trim(path), exist=res)
```

Comment:
It seems to appear that ifort will set `res` to `.false.` on the above `inquire` call, but gfortran does not.

Reply:
This check may not really be necessary. The check in the next code block on whether the `index` file exists is a sufficient condition for the directory existing.

Reply:
Actually, taking a look at the standard, it doesn't appear that `inquire` is able to work on directories. From section 9.9 of the Fortran 2003 standard:

> The INQUIRE statement may be used to inquire about properties of a particular named file or of the
> 25 connection to a particular unit.

And then in 9.4:

> A unit, specified by an io-unit, provides a means for referring to a file.

So it looks like gfortran's ability to inquire on a directory is an extension, similarly to ifort's `directory` keyword.


I also don't think this `inquire` statement is really needed, as the next `inquire` (into the index file) will return `.false.` if the directory doesn't exist. So I think its safe to remove it.